catroid/src/main/java/org/catrobat/catroid/stage/PreStageActivity.java

```diff
@@ -563,6 +595,31 @@ private void nfcInitialize() {
 		resourceInitialized();
 	}
 
+	private static ArrayList<Brick> getBricksRequieringResource(int resource) {
+		ArrayList<Brick> brickList = new ArrayList<Brick>();
+		ArrayList<Sprite> spriteList = (ArrayList<Sprite>) ProjectManager.getInstance().getCurrentProject()
+				.getSpriteListWithClones();
+
+		for (Sprite sprite : spriteList) {
+			brickList.addAll(sprite.getBricksRequiringResource(resource));
+		}
+
+		//unique entries
+		Iterator<Brick> i = brickList.iterator();
+		while (i.hasNext()) {
+			Brick c = i.next();
+			Iterator<Brick> secIt = brickList.iterator();
+			while (secIt.hasNext()) {
+				Brick proofBrick = secIt.next();
+				if (proofBrick.getClass() == c.getClass() && proofBrick != c) {
+					i.remove();
+					break;
+				}
+			}
+		}
```

Comment:
Please take a look at the android utils stuff and guava. Maybe they provide something useful.
Let me know if the result has to be sorted or not.